cmd/auth-rpc-client.go

```diff
@@ -80,13 +80,13 @@ func newAuthRPCClient(config authConfig) *AuthRPCClient {
 
 // Login - a jwt based authentication is performed with rpc server.
 func (authClient *AuthRPCClient) Login() (err error) {
-	authClient.Lock()
-	defer authClient.Unlock()
-
+	authClient.RLock()
 	// Return if already logged in.
 	if authClient.authToken != "" {
+		authClient.RUnlock()
 		return nil
 	}
+	authClient.RUnlock()
```

Comment:
Now I see the `Lock()` is moved above `authClient.rpcClient.Call(loginMethod, &loginArgs, &loginReply);`.  This doesn't prevent parallel logins either ie the first thread is in process of login and other thread will do login once `Unlock()`.  I am still favor of the previous approach wrt `Login()`.

As you defer, I would recommend someone needs to give +1 wrt `Login()`.  Other changes are good to me.